zerver/migrations/0379_userprofile_uuid_backfill.py

```diff
@@ -0,0 +1,40 @@
+import uuid
+
+from django.db import migrations, models
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def backfill_user_profile_uuid(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+
+    max_id = UserProfile.objects.aggregate(models.Max("id"))["id__max"]
+    if max_id is None:
+        # Nothing to do if there are no users yet.
+        return
+
+    BATCH_SIZE = 10000
+    lower_bound = 1
+
+    while lower_bound <= max_id:
+        user_profiles_to_update = []
+        for user_profile in UserProfile.objects.filter(
+            id__gte=lower_bound, id__lte=lower_bound + BATCH_SIZE, uuid=None
+        ).only("id", "uuid"):
+            user_profile.uuid = uuid.uuid4()
+            user_profiles_to_update.append(user_profile)
+        lower_bound += BATCH_SIZE + 1
```

Comment:
We generally prefer to write this sort of migration where we're adding BATCH_SIZE and making the loops non-inclusive on one side; it's just easier to read that way.

I think one can implement that by making it `id_gt=lower_bound`, and having `lower_bound` start as 0.

Reply:
To be explicit, you also need to adjust so `lower_bound += BATCH_SIZE` not `BATCH_SIZE + 1`, as well.  Otherwise you do 0 - 10000 inclusive in the first batch, then 10002 to 20001, then 20003 to 30002, ...

Reply:
```
dff --git a/zerver/migrations/0379_userprofile_uuid_backfill.py b/zerver/migrations/0379_userprofile_uuid_backfill.py
index 792dc562af..9750f6b862 100644
--- a/zerver/migrations/0379_userprofile_uuid_backfill.py
+++ b/zerver/migrations/0379_userprofile_uuid_backfill.py
@@ -14,16 +14,16 @@ def backfill_user_profile_uuid(apps: StateApps, schema_editor: DatabaseSchemaEdi
         return
 
     BATCH_SIZE = 10000
-    lower_bound = 1
+    lower_bound = 0
 
-    while lower_bound <= max_id:
+    while lower_bound < max_id:
         user_profiles_to_update = []
         for user_profile in UserProfile.objects.filter(
-            id__gte=lower_bound, id__lte=lower_bound + BATCH_SIZE, uuid=None
+            id__gt=lower_bound, id__lte=lower_bound + BATCH_SIZE, uuid=None
         ).only("id", "uuid"):
             user_profile.uuid = uuid.uuid4()
             user_profiles_to_update.append(user_profile)
-        lower_bound += BATCH_SIZE + 1
+        lower_bound += BATCH_SIZE
 
         UserProfile.objects.bulk_update(user_profiles_to_update, ["uuid"])
 
```

Please check if that looks right, these always feels easy to trip up on

Reply:
Looks right to me!